### DISCUSSION TITLE
NextLink Component doesnt work when I use middleware in Next.js 13?

### DISCUSSION THREAD

**Discussion by tofuwu-dev** [vercel/next.js#55265] [Help]
### Summary

Hi everyone,

Now I am implementing i18n in nextjs app router.
I am using middlerware to redirect like - `/en/profile` or `/de/profile`.
But after that, next link doesnt work as I expected, it is hard-refreshing whenever I navigate page.

How to solve this issue? Thanks!

**Comment by LimSB-dev** [comment]
Without specific code examples and more details about your setup, it's challenging to pinpoint the exact issue. Please provide more information about your routing and i18n configuration, and I'd be happy to offer more specific guidance.

**Reply by tofuwu-dev** [reply]
Hi @LimSB-dev 

Thanks for your reply and sorry for my tardy response!

I think I had a issue in another part on my end. Before I fix that issue, it seemed Next Link componenet dont work as middleware. Now it's fine!

But I have one more question with this chance.

I think middleware doesnt work in firebase hosting for ssg mode. Is it issue from firebase or nextjs?

**Reply by LimSB-dev** [reply]
First of all, it will be difficult to give you an accurate answer without detailed information and code, but the following is the information that will help you as soon as you think of it.
1. Check your Next.js middleware implementation to ensure it's set up correctly and compatible with SSG mode.

2. Verify that the middleware works as expected in a local development environment with Next.js.

3. Review Firebase Hosting configuration to make sure you're not missing any settings or routes.

**Reply by tofuwu-dev** [reply]
@LimSB-dev middleware doesnt work for ssg mode.
